### DISCUSSION TITLE
How measure CPU load of lpc55s16 microcontroller

### DISCUSSION THREAD

**Discussion by aravinth-kalai** [zephyrproject-rtos/zephyr#71647] [Q&A] [answered]
Hello everyone, 

Currently I'm trying to measure CPU load of lpc55s16, Initially I tried to measure the CPU load by using the formula 

((Total_thread_execution_time / total_CPU_up_time)*100)

using the k_uptime_get() API, but unfortunately k_uptime_get() is not accurate so my calculation is not working as expected, so is there any dedicated API is available in zephyr to measure CPU load? if yes please share that will be really helpful.

Thanks in advance!

**Comment by aravinth-kalai** [comment]
Hi @DerekSnell could you please help me on this issue.

**Comment by DerekSnell** [comment]
Hi @aravinth-kalai ,
I appears to me that with `CONFIG_SCHED_THREAD_USAGE=y` selected, you can call `k_thread_runtime_stats_all_get()` to get the CPU runtime stats.  You can read about [k_thread_runtime_stats](https://github.com/zephyrproject-rtos/zephyr/blob/eddb3822d2aa746026837252b3e337b0c4ef067a/include/zephyr/kernel/thread.h#L207) for the values in that struct.  And then I believe the CPU load would be `execution_cycles` / `total_cycles`.

Let us know how that works for you.  Best regards

**Comment by aravinth-kalai** [comment, marked as answer]
Hi @DerekSnell , 

Thanks for your reply, After searching a lot finally I got [this ](https://docs.zephyrproject.org/latest/services/debugging/thread-analyzer.html) thread analyzer feature in zephyr that helps me to print CPU load of each thread and ideal thread.
